README.md

```diff
@@ -31,43 +31,56 @@ When connecting to a PostgreSQL database running on a Windows host from a WSL2 i
+ - Root folder: API routes (in `main.py`), schedule definition and other core dependencies, i.e. PostgreSQL, MongoDB and Redis
+ - `tables`: Each table, or set of very similar tables (like tags) has a module. This contains all of the SQL for modifying the data and structure of the table, so even `migrations` call out to functions in these modules. The `tables` modules will contain limited business logic, mostly where this would be the equivalent of a database trigger
+ - `migrations`: These manage database schema updates. Note in most cases the SQL itself is in the relevant `tables` module. The migration history is stored in a table called `mikro_orm_migrations` for backward compatibility with the previous NestJS / Mikro-ORM implementation of this project
+ - `services`: This is where most of the complex business logic lives, where interactions between multiple tables are coordinated
+ - `models`: These are the [Pydantic](https://docs.pydantic.dev/) classes used to validate API requests and generate OpenAPI documentation. Note that Pydantic models are only used for outward-facing models. Internally, data is mainly passed around using dictionaries or the [asyncpg Record](https://magicstack.github.io/asyncpg/current/api/index.html#asyncpg.Record) structure (which behaves like a dictionary)
+ - `views`: The SQL definition for any views that are created in the database during migrations
+ - `assets`: Non-python resources
 
-The domain/entities folder defines the main entities and is used to automatically generate migrations using the [migration:create](https://mikro-orm.io/docs/migrations#initial-migration) npm task.
+The entrypoint is main.py which runs database migrations and starts the service and scheduler.
 
 ## Testing
 
-The unit tests use testcontainers to create a temporary Postgres database in Docker, which lasts for the duration of the test run. The tests share the same database while running, so ensure that tests are independant from one another by using randmoised product codes / tags.
+The unit tests use testcontainers to create a temporary Postgres database and Redis instance in Docker, which lasts for the duration of the test run. The tests share the same database while running, so ensure that tests are independent from one another by using randomized product codes / tags.
+
+Tests are mingled in with the project structure to make it easier to find them. A TDD approach to development is recommended.
 
 ## Calling from Product Opener
 
-By default, product opener is configured to call the "query" host on the "po_default" network. To configure Product Opener to use a locally running instance update the following line in the Product Opener .env file:
+By default, product opener is configured to call the "query" host on the `COMMON_NET_NAME` network. To configure Product Opener to use a locally running instance update the following line in the Product Opener .env file:
 
 ```
 QUERY_URL=http://host.docker.internal:5510
```

Comment:
I think this only work on Mac.

Reply:
No, it should also work on other platforms because we have an `extra_hosts` entry of `"host.docker.internal:host-gateway"` in the dev.yml docker compose file for product opener.